gravityforms-openai/gfoai-save-prompt-to-entry.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Gravity Wiz // Gravity Forms OpenAI // Save Prompt to Entry
+ *
+ * Sometimes, a prompt could have a lot of conditionals, so you can't simply copy it using a
+ * Live Merge Tag. This snippet grabs the final prompt that's ultimately sent to OpenAI and
+ * stores it along with the entry.
+ *
+ * Instructions:
+ *  1. Install per https://gravitywiz.com/how-do-i-install-a-snippet/
+ */
+
+// Update to the targetted form id, and the field id of the hidden field on form where we store the prompt.
+$form_id  = '123';
+$field_id = '4';
+
+add_filter( 'gf_openai_request_body', function ( $body, $endpoint, $feed ) use ( $form_id, $field_id ) {
+	if ( ! $field_id || ( $form_id && $feed['form_id'] != $form_id ) ) {
+		return $body;
+	}
+
+	add_action( 'gform_post_process_feed', function ( $_feed, $entry ) use ( $body, $feed, $field_id ) {
+		if ( $_feed['id'] == $feed['id'] ) {
+			$entry[ $field_id ] = $body['messages'][0]['content'];
```

Comment:
How does this work? Is `$entry` passed as a reference?

Reply:
sorry about that. Updated. Please see now!